docs/design/README.md

```diff
@@ -0,0 +1,552 @@
+
+Briefly, the entry point of the existing build system is Rust's build system, `cargo`.
+A plugin called [cargo make] gets its instructions from `Makefile.toml`.
+We use `cargo` to resolve dependencies between variants and packages, and we use custom build scripts to launch docker-hosted build environments.
+Packages are defined and built using RPM.
+
+These package builds take place in a docker environment using `docker build` commands and `buildx`.
+The toolchains used to compile Rust, C, C++ and Go software are distributed in a container image called the [bottlerocket-sdk].
+It is in the SDK environment that the package builds take place.
+
+Once all the software for a Bottlerocket image has been built into RPM images, a final image creation step occurs.
+A `docker build` command installs everything into a rooted image file.
+
+The toolchains necessary for each arch (x86_64 and Arm) are available as separate SDK containers to allow for cross-compilation.
+The build host has the following requirements: sufficient versions of Docker, buildx, Cargo, Cargo Make and a few tools like lz4 and jq.
+The root of the Bottlerocket git repo is mounted and artifacts are created in `.cargo` and `build` directories.
+
+Certain custom tools for building, testing and publishing Bottlerocket (buildsys, testsys, and pubsys) are compiled from source during a Bottlerocket build.
+These are built using the host-installed Rust toolchain instead and not the SDK.
+Once these are built, they are used elsewhere in the build process.
+Buildsys, for example, is the tool that launches container build environments from `build.rs` scripts.
+
+[How the Bottlerocket build system works]: https://aws.amazon.com/blogs/opensource/how-the-bottlerocket-build-system-works/
+[cargo make]: https://github.com/sagiegurari/cargo-make
+[bottlerocket-sdk]: https://github.com/bottlerocket-os/bottlerocket-sdk
+
+### Enhancing the Build System
+
+The existing build system will remain intact with certain features and abstractions added to it.
+Twoliter will serve as a facade providing a new interface over the existing build system.
+Twoliter will understand how to start a Bottlerocket build using the OOTB user's sources, packages etc.
+It will inject these directories into the existing build system by setting existing and new `Makefile.toml` variables.
+
+Twoliter will eliminate the need for all build-host software other than `docker.`
+A bootstrapping container will be published that contains any software needed to kick off and sustain a `cargo make` build command.
+
+## Kits
+
+A central idea in this design is that the build system will be able to produce and consume Docker images that contain RPM packages.
+We call these images *kits*.
+A kit is a Yum repo distributed in its entirety, atomically, as a container image.
+Kits will satisfy requirements 4-8 above.
+Existing container registry mechanisms, such as ECR, will automatically sign these images satisfying requirement 1.
```

Comment:
Two points here:
* container signing isn't automatic, client tooling is needed
* it would be nice if RPMs were GPG-signed and if `dnf` enforced signature checking for the non-local kit repos